### PR TITLE
NEI item manager overlay integration

### PR DESCRIPTION
**Pull request by ghostflyby**
Previously, the NEI asm did not work as `GuiCraftGUI` override some methods and treat Slots specially as storage logic only,
 so I do Hooks manually and add a `IContainerObjectHandler ` to handle the slot under mouse.

Other fixes might be 
1. mixin, but it’s bad mixining everywhere
2. refactor NEI for more general treatments, which is no better than mixin as it’s based on asm.

截屏2023-09-25 21 40 22

### CODE REVIEW

**Comment by ghostflyby** [comment]
The CI failed on … build env setup?

**Comment by miozune** [comment]
Updating buildscript should fix it

**Comment by Connor-Colenso** [comment]
I was just thinking about this the other day. Will be great to solve this long standing issue!

File: src/main/java/binnie/core/craftgui/minecraft/GuiCraftGUI.java [hunk 1]
```diff
@@ -506,4 +523,11 @@ public void drawRect(float x, float y, float width, float height, int color) {
         GL11.glDisable(GL11.GL_BLEND);
         GL11.glEnable(GL11.GL_TEXTURE_2D);
     }
+
+    public ItemStack getStackUnderMouse(int x, int y) {
+        Slot slot = window.calculateMousedOverWidgets().stream().filter(widget -> widget instanceof ControlSlot)
+                .findFirst().map(w -> ((ControlSlot) w).slot).orElse(null);
```
Comment (miozune) [inline src/main/java/binnie/core/craftgui/minecraft/GuiCraftGUI.java:529]: You can chain methods like `.map(w -> ((ControlSlot) w).slot).map(Slot::getStack).orElse(null)`

**Review by miozune — COMMENTED** [review]
Drag-and-drop like bookmark reorder doesn't work

**Comment by ghostflyby** [comment]
> Drag-and-drop like bookmark reorder doesn't work

It should work now

**Review by miozune — APPROVED** [review]

**Pull request merged**